validator/main.c

```diff
@@ -0,0 +1,44 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include "standalone/api.h"
+
+static void print_log(void* obj, int level, char* str)
+{
+    if (str != NULL) {
+        puts(str);
+    }
+}
+
+static void exit_with_error(const char* msg)
+{
+    puts(msg);
+    exit(EXIT_FAILURE);
+}
+
+int main(int argc, char* argv[])
+{
+    if (argc != 2) {
+        exit_with_error("Invalid number of command line arguments");
+    }
+
+    server_rec* modsec_server = modsecInit();
+    if (modsec_server == NULL) {
+        exit_with_error("Failed to create ModSecurity server structure");        
+    }
+    char hostname[] = "localhost";
+    modsec_server->server_hostname = hostname;
+
+    modsecSetLogHook(NULL, print_log);
```

Comment:
all these function will be removed out of the original modsec code in the future?

Reply:
No, they won't, why would they?

Reply:
so these code will be run twice if tenant call the main function and start modsec?

Reply:
Yes. The standalone utility is executed once for every ModSecurity configuration by the tenant. Then Nginx will also use these functions to initialize configuration for the datapath.